### DISCUSSION TITLE
Help Maintain

### DISCUSSION THREAD

**Discussion by nyurik** [maplibre/maplibre-gl-js#1] [General]
This repo had to be recreated, copy/pasting the original text from @HarelM and :+1: by @kaplanlior 

---

Hi,

I'll be happy to help out in any way that I can.
I have sent a pull request to the original repository to facilitate for custom tiles source (i.e. offline sqlite in my case).
I'll be happy if this lands here so I won't need to maintain my fork.

Feel free to tag me @HarelM.
My app and site relay heavily on mapbox-gl-js but not on their tiles and data and I never used a token for that reason.
Also my site is open source, so storing a token there is a bit problematic in theory, but not the real reason I'm here...

Hope I can be of assistance.

**Comment by markgis** [comment]
I'd happily help where I can as well, was thinking as a relative new comer to the open world, I could help with documentation?

Any thoughts on using Docbox? Have used that lib before, I can grab some of the main docs over from Mapbox and start creating basic `First steps` docs? Or is it easier to just use the Git wiki pages?

**Reply by geoapify** [reply]
Also ready to help at least with documentation. Are there plans already for that? I believe that the original documentation is copyright protected. However, there are typescript definitions that are under MIT license (@types/mapbox-gl@1.13.0). This can be used to generate .html or .md documentation (for example, with typedoc)  that can be extended later.

**Comment by HarelM** [comment]
I'm also using the following great library with my Angular project (also applies to ionic):
https://github.com/Wykks/ngx-mapbox-gl/issues/260
I think it might be interesting to host it here as well assuming there's an interest of the developer to migrate.
I can open a new discussion on this as it is only half relevant to this discussion, but since there's a dependency on mapbox there it might also need a fork here with the relevant changes to dependencies...

**Reply by aprilmay** [reply]
Would this be a ngx-maplibre-gl project? Any update?

**Reply by HarelM** [reply]
I'm still not using maplibre since I'm looking for this issue to be merged:  https://github.com/maplibre/maplibre-gl-js/issues/29 Once this is integrated into this library I'll be able to test maplibre and after that is successful I'll be sure to fork the angular wrapper to use this project. @aprilmay If you want to push this forward yourself please do, I'm not a maintainer of either project unfortunately and the guys at ngx-mapbox-gl do not want to migrate as far as I understand from this thread: https://github.com/Wykks/ngx-mapbox-gl/issues/260

**Reply by wipfli** [reply]
Crossreading #29 is quite interesting. It's the first time I had a look at it. @HarelM would you agree with the one sentence summary "#29 implements offline map rendering"?

**Reply by HarelM** [reply]
@wipfli Yes, and so much more. The ability to implement the code that happens when the renderer is requesting a tile allow you to even create in memory tiles from data stored anywhere in any format. It is, from my point of view, the equivalent of the "loadTileFunction" implemented in openlayers (or at least something very close to it).
In any case, I'm still waiting for it to be merged, I'm already using this with a `patch-package` over mapbox-gl code to allow offline tiles for a Cordova app, so I know this works and allows offline map rendering :-) I'm just waiting for this to be merged and released in order to migrate my code base to use maplibre-gl...

**Reply by HarelM** [reply]
@nyurik @wipfli @lseelenbinder I would like to start my migration to maplibre. I have changed the references in my code to use maplibre but it seems that the references in ngx-mapbox-gl still exists, so I need to fork and migrate ngx-mapbox-gl now. I would like it to be hosted in this github company and would like to maintain it, or at least try to. Can you help me with this task by either granting me the right permissions or forking the repository so I can start opening pull requests?
ngx-mapbox-gl (will be ngx-maplibre-gl) is an angular wrapper around mapbox- gl (maplibre-gl) that allows good binding between angular and mapbox (maplibre).

**Comment by vinayakkulkarni** [comment]
We've created a fork of [`v-mapbox`](https://github.com/geospoc/v-mapbox) and we (GeoSpoc) are planning to maintain it as a pure OSS MIT-licensed based version

**Reply by HarelM** [reply]
I think it will be interesting to add somewhere in the documentation a note about wrappers that are maintained by the community such as the two above assuming they'll be using this fork of mapbox.
Better even if all of them are hosted in this github company, but it's only an idea. I know that some might not want to host the repo here...
